static/app/components/profiling/frameStack.tsx

```diff
@@ -34,36 +35,125 @@ function formatColorForFrame(
   return `rgba(${color.map(n => n * 255).join(',')}, 1.0)`;
 }
 
+function makeSortFunction(
+  property: 'total weight' | 'self weight' | 'name',
+  direction: 'asc' | 'desc'
+) {
+  if (property === 'total weight') {
+    return direction === 'desc'
+      ? (a: FlamegraphFrame, b: FlamegraphFrame) => {
+          return b.node.totalWeight - a.node.totalWeight;
+        }
+      : (a: FlamegraphFrame, b: FlamegraphFrame) => {
+          return a.node.totalWeight - b.node.totalWeight;
+        };
+  }
+
+  if (property === 'self weight') {
+    return direction === 'desc'
+      ? (a: FlamegraphFrame, b: FlamegraphFrame) => {
+          return b.node.selfWeight - a.node.selfWeight;
+        }
+      : (a: FlamegraphFrame, b: FlamegraphFrame) => {
+          return a.node.selfWeight - b.node.selfWeight;
+        };
+  }
+
+  if (property === 'name') {
+    return direction === 'desc'
+      ? (a: FlamegraphFrame, b: FlamegraphFrame) => {
+          return a.frame.name.localeCompare(b.frame.name);
+        }
+      : (a: FlamegraphFrame, b: FlamegraphFrame) => {
+          return b.frame.name.localeCompare(a.frame.name);
+        };
+  }
+
+  throw new Error(`Unknown sort property ${property}`);
+}
+
 interface FrameCallTreeStackProps {
   flamegraphRenderer: FlamegraphRenderer;
+  referenceNode: FlamegraphFrame;
   roots: FlamegraphFrame[];
 }
 
-function FrameCallTreeStack({flamegraphRenderer, roots}: FrameCallTreeStackProps) {
+function FrameCallTreeStack({
+  flamegraphRenderer,
+  roots,
+  referenceNode,
+}: FrameCallTreeStackProps) {
+  const [sort, setSort] = useState<'total weight' | 'self weight' | 'name' | null>(
+    'total weight'
+  );
```

Comment:
Why does this need `null`?

Reply:
Is the default here not `total weight`? I don't see any place that will set this to `null`

Reply:
I'll check, I need to cross verify what some tools do here, often times when you are looking at inverted, it's default sorted by self weight and when its call order, you'll get by total weight. I'll try setup some sensible defaults and look into removing the null